crates/runtime/src/model.rs

```diff
@@ -14,137 +14,38 @@ See the License for the specific language governing permissions and
-    },
-
-    #[snafu(display("Unable to run model: {source}"))]
-    UnableToRunModel { source: crate::modelruntime::Error },
-
-    #[snafu(display("Unable to load required secrets"))]
-    UnableToLoadRequiredSecrets {},
-}
-
-impl Model {
-    pub async fn load(
-        model: spicepod::component::model::Model,
-        secret: Option<Secret>,
-    ) -> Result<Self> {
-        let source = source(&model.from);
-        let source = source.as_str();
-
-        let Some(secret) = secret else {
-            tracing::warn!(
-                "Unable to load model {}: unable to get secret for source {}",
-                model.name,
-                source
-            );
-            return UnableToLoadRequiredSecretsSnafu {}.fail();
-        };
-
-        let mut params = std::collections::HashMap::new();
-        params.insert("name".to_string(), model.name.to_string());
-        params.insert("path".to_string(), path(&model.from));
-        params.insert("from".to_string(), path(&model.from));
-        params.insert("files".to_string(), model.files.join(",").to_string());
-
-        let tract = crate::modelruntime::tract::Tract {
-            path: create_source_from(source)
-                .context(UnknownModelSourceSnafu)?
-                .pull(secret, Arc::new(Option::from(params)))
-                .await
-                .context(UnableToLoadModelSnafu)?
-                .clone()
-                .to_string(),
-        }
-        .load()
-        .context(UnableToInitModelSnafu {})?;
-
-        Ok(Self {
-            runnable: tract,
-            model: model.clone(),
-        })
-    }
-
-    pub async fn run(&self, df: Arc<RwLock<DataFusion>>) -> Result<RecordBatch> {
-        let data = df
-            .read()
-            .await
-            .ctx
-            .sql(
-                &(format!(
-                    "select * from datafusion.public.{} order by ts asc",
-                    self.model.datasets[0]
-                )),
-            )
-            .await
-            .context(UnableToQuerySnafu {})?
-            .collect()
-            .await
-            .context(UnableToQuerySnafu {})?;
-
-        let result = self.runnable.run(data).context(UnableToRunModelSnafu {})?;
-
-        Ok(result)
-    }
-}
-
-#[must_use]
-pub(crate) fn source(from: &str) -> String {
-    match from {
-        s if s.starts_with("spiceai:") => "spiceai".to_string(),
-        s if s.starts_with("huggingface:") => "huggingface".to_string(),
-        s if s.starts_with("file:/") => "localhost".to_string(),
-        _ => "spiceai".to_string(),
-    }
-}
-
-#[must_use]
-pub(crate) fn path(from: &str) -> String {
-    let sources = vec!["spiceai:"];
+use crate::DataFusion;
 
-    for source in &sources {
-        if from.starts_with(source) {
-            match from.find(':') {
-                Some(index) => return from[index + 1..].to_string(),
-                None => return from.to_string(),
-            }
-        }
+pub async fn run(
```

Comment:
Convient function for `runtime` crate so that the `models` crate can continue to be ignorant of Datafusion.